tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -4406,6 +4405,202 @@ TEST_F(InputProcessorFixture, epJSONgetObjectItem_minfields)
     EXPECT_NEAR(state->dataIPShortCut->rNumericArgs(4), 0.0, 0.0001);
 }
 
+TEST_F(InputProcessorFixture, epJSONgetFieldValue_fromJSON)
+{
+
+    json root;
+    std::string obj_type1 = "Building";
+    std::string name1 = "Building 1";
+    json bldg1 = {{"loads_convergence_tolerance_value", 0.1}, {"terrain", "Ocean"}};
+    EXPECT_TRUE(bldg1.is_object());
+    root[obj_type1][name1] = bldg1;
+
+    std::string obj_type2 = "Material";
+    std::string name2 = "Standard insulation_01";
+    json mat1 = {{"name", name1}, {"roughness", "MediumRough"}, {"thickness", 0.2}, {"solar_absorptance", 0.5}};
+    EXPECT_TRUE(mat1.is_object());
+    root[obj_type2][name2] = mat1;
+
+    state->dataInputProcessing->inputProcessor->epJSON = root;
+
+    state->dataGlobal->isEpJSON = true;
+    state->dataInputProcessing->inputProcessor->initializeMaps();
+    std::string alphaFieldValue;
+    Real64 numericFieldValue = 0.0;
+    json objectSchemaProps;
+
+    // Building object
+    objectSchemaProps = state->dataInputProcessing->inputProcessor->getObjectSchemaProps(*state, obj_type1);
+    // User inputs from above
+    // Note even though choice keys are case-sensitive during epJSON processing, getFieldValue pushes Alphas to UPPERcase
```

Comment:
How is retain case handled?

            "group": "Simulation Parameters",
            "name": {
                "type": "string",
                "retaincase": true,
                "default": "NONE"
            },